java_src/foam/dao/PMDAO.java

```diff
@@ -0,0 +1,21 @@
+package foam.dao;
+
+import foam.core.FObject;
+import foam.nanos.pm.PM;
+import foam.nanos.pm.PMInfo;
+
+/**
+ * Created by nick on 19/05/17.
+ */
+public class PMDAO extends ProxyDAO {
+
+  @Override
+  public FObject put(FObject obj) {
+    PMInfo pmi = (PMInfo)obj;
+    PM pm = new PM(PMDAO.class, pmi.getClsname() + ":" + pmi.getPmname());
+    getDelegate().put(obj);
+    pm.end(getX());
```

Comment:
Put the end() call (which should be renamed to log()) into the finally block of a try/finally so that the PM isn't lost if the delegate put() throws an exception.